articles/coding-on-purpose.md

```diff
@@ -0,0 +1,99 @@
+
+> By searching for solutions we restrict ourselves with constraints which don't exist in the original problem.
+
+When you're stuck in such a situation it's worth taking one or two steps back and asking: *what again was the goal we wanted to achieve?* And then again: *why did we want to achieve this goal in the first place*? This helps you opening up the space for possible solutions and hopefully let's you discover that the problem you're trying to solve does not even need solving but instead you find a simpler, less problematic solution to the initial *why*. If you ask *why* five times, this approach even has a name: [The five whys technique](https://en.wikipedia.org/wiki/5_Whys).
+
+### Measure Outcome, Not Output
+
+Think about how you track achievement of desired outcome in your project. In the end, it's the job of your *Product Owner* or *Product Manager*, but if you think they could use some inspiration, they are probably greatful for some ideas. As an example, [John Cutler proposes](https://twitter.com/johncutlefish/status/775523444385382400) to add a column to your task board called *Achieved desired outcome*. Or [Gregor Ilg says](https://twitter.com/_gregorilg/status/884663883172110336):
+
+> Don't celebrate when you launched a product. Celebrate once you've learned from it.
+
+Instead of measuring how many story points we can deliver per sprint, we should apply our math skills much more to measuring and tracking whether we achieve the desired outcomes in production. Do we get the desired user adoption? Is the desired percentage of business done on the new platform? Do we save the desired amount of manual work using the new system?
+
+### Empathize With The People That Matter
+
+Empathy and good collaboration with the end users is of course important in order to build a good user experience. But listening to the users is not enough.
+
+You might hear a lot of different goals from different people in the project. Especially if there's a lack of official goals, people might fill them up with their own hidden or not so hidden agenda. Try to figure out who spends money for the project and try to understand what outcomes they hope to achieve through this investment.
+
+### Lead Through Purpose
+
+Managers should put a strong focus on communicating the *why* of the things they ask you or a team to do and also judge your success by whether the *why* is fulfilled and not whether you built exactly *what* they asked for and did it *how* they expected you to do it. Simon Sinek would agree that communicating the *why* effectively increases the engagement of the team.
+
+### Coding Tips
+
+* If you are unsure whether what you are currently working on is actually needed for something meaningful, ask the relevant person (e.g. product owner) for the purpose of your work. It might turn out that it's not so important after all and that the stress you are experiencing is actually artificial and does not benefit anybody.

```

Comment:
Hat für mich die selbe Message wie "When The Solution Becomes The Problem"
-> Aussage mehr differenzieren oder dann weglassen?